index.go

```diff
@@ -403,26 +402,42 @@ func (u *UintFieldIndex) FromArgs(args ...interface{}) ([]byte, error) {
 	}
 
 	val := v.Uint()
-	buf := make([]byte, size)
-	binary.PutUvarint(buf, val)
+	buf := encodeUInt(val, size)
 
 	return buf, nil
 }
 
+func encodeUInt(val uint64, size int) []byte {
+	buf := make([]byte, size)
+
+	switch size {
+	case 1:
+		buf[0] = uint8(val)
+	case 2:
+		binary.BigEndian.PutUint16(buf, uint16(val))
+	case 4:
+		binary.BigEndian.PutUint32(buf, uint32(val))
+	case 8:
+		binary.BigEndian.PutUint64(buf, val)
+	}
+
+	return buf
+}
+
 // IsUintType returns whether the passed type is a type of uint and the number
 // of bytes needed to encode the type.
 func IsUintType(k reflect.Kind) (size int, okay bool) {
 	switch k {
 	case reflect.Uint:
-		return binary.MaxVarintLen64, true
+		return 8, true
```

Comment:
_Technically_ this is platform dependent. I don't think that's a huge deal as 32 bit platforms will still work but just waste 4 bytes of allocated memory.

But it seems like this would be safer if it used `bits.UintSize`. What do you think?

```suggestion
		return bits.UintSize, true
```

Reply:
Fixed. `bits.UintSize` is the number of bits, so I had to divide it by 8. I don't see a constant in go for the number of bits per byte. Do you know of one?

Reply:
I don't, I think this sounds good though! Thanks.